lib/paper_trail/reifier.rb

```diff
@@ -91,14 +91,17 @@ def init_unversioned_attrs(attrs, model)
       #
       # @api private
       def reify_attribute(k, v, model, version)
-        if model.has_attribute?(k)
-          model[k.to_sym] = v
-        elsif model.respond_to?("#{k}=")
-          model.send("#{k}=", v)
-        elsif version.logger
-          version.logger.warn(
-            "Attribute #{k} does not exist on #{version.item_type} (Version id: #{version.id})."
-          )
+        begin
+          if model.has_attribute?(k)
+            model[k.to_sym] = v
+          elsif model.respond_to?("#{k}=")
+              model.send("#{k}=", v)
+          elsif version.logger
+            version.logger.warn(
+              "Attribute #{k} does not exist on #{version.item_type} (Version id: #{version.id})."
+            )
+          end
+        rescue ActiveRecord::ReadonlyAttributeError
```

Comment:
Found where this was implemented in https://github.com/rails/rails/pull/46105/files#diff-55ee06def7a48911bcbe1a6204576b805a842db2bac44ac66d1732d8c1f44db0R4  ... and doesn't exist before 7.1 unfortunately 😞 

https://github.com/rails/rails/blob/1060cec88bfc5d45a6563d6966f6cb7533057b2b/activerecord/lib/active_record/readonly_attributes.rb#L3-L9

So we need a different way to handle it 🤔  Maybe a `handle_readonly_error` helper that takes a block, except it's defined to yield rescue this only on 7.1 (or when it's defined?), otherwise it's an no-op to just yield.

Reply:
Thank you for this feedback, I've updated the code.